### PR TITLE
Lift API extension

### PR DESCRIPTION
**Pull request by microsievert**
Added ``ChangeLock()`` method. Old version of method was not work.

### CODE REVIEW

File: Exiled.API/Features/Lift.cs [hunk 1]
```diff
@@ -236,6 +232,39 @@ public static bool TryMeltPlayer(Player player)
...
+        public void ChangeLock(DoorLockReason lockReason)
+        {
+            DoorLockReason lockFlags = Base.ActiveLocks;
+
+            if (lockReason == DoorLockReason.None)
+                lockFlags = 0;
+            else
+                lockFlags |= lockReason;
+
+            List<ElevatorDoor> targetDoors = ElevatorDoor.AllElevatorDoors.First(x => x.Key == Base.AssignedGroup).Value;
```
Comment (iopietro) [inline Exiled.API/Features/Lift.cs:248]: This should become a property that gets populated with Exiled doors in the constructor, something like `IReadOnlyCollection<Door> Doors`.

File: Exiled.API/Features/Lift.cs [hunk 2]
```diff
@@ -236,6 +232,39 @@ public static bool TryMeltPlayer(Player player)
...
+
+            foreach (ElevatorDoor door in targetDoors)
+            {
+                if (lockFlags == 0)
```
Comment (iopietro) [inline Exiled.API/Features/Lift.cs:252]: You should use the in-game `DoorVariant::ServerChangeLock(DoorLockReason, bool)` method.

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Lift.cs (hunk 2 above)
Comment (microsievert) [reply Exiled.API/Features/Lift.cs]: Solved

File: Exiled.API/Features/Lift.cs [hunk 3]
```diff
@@ -236,6 +245,36 @@ public static bool TryMeltPlayer(Player player)
...
+        /// Changes lock of the lift.
+        /// </summary>
+        /// <param name="lockReason">Type of lift lockdown.</param>
+        public void ChangeLock(DoorLockReason lockReason)
+        {
+            bool forceLock = lockReason != DoorLockReason.None;
+
+            List<ElevatorDoor> targetDoors = ElevatorDoor.AllElevatorDoors.First(x => x.Key == Base.AssignedGroup).Value;
+
+            foreach (ElevatorDoor door in targetDoors)
```
Comment (iopietro) [inline Exiled.API/Features/Lift.cs:258]: You should use the new `Doors` property.

File: Exiled.API/Features/Lift.cs [hunk 4]
```diff
@@ -236,6 +245,36 @@ public static bool TryMeltPlayer(Player player)
...
+                {
+                    door.NetworkActiveLocks = 0;
+
+                    door.ServerChangeLock(DoorLockReason.None, true);
+                }
+                else
+                {
+                    door.ServerChangeLock(lockReason, true);
+
+                    if (Base.CurrentLevel != 1)
```
Comment (iopietro) [inline Exiled.API/Features/Lift.cs:270]: You should directly use the `Lift::CurrentLevel` property.

**Review by iopietro — CHANGES_REQUESTED** [review]

**Pull request merged**